src/main/java/org/generationcp/middleware/liquibase/UpdateLabelPrintingPresetsTask.java

```diff
@@ -0,0 +1,130 @@
+import org.generationcp.middleware.domain.labelprinting.LabelPrintingPresetDTO;
+import org.generationcp.middleware.domain.labelprinting.PresetDTO;
+
+import java.io.IOException;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.stream.Collectors;
+
+public class UpdateLabelPrintingPresetsTask implements liquibase.change.custom.CustomTaskChange {
+
+	private static final Integer MAX_FIELD_STATIC_ID = 63;
+	private ObjectMapper jacksonMapper;
+
+	@Override
+	public void execute(final Database database) throws CustomChangeException {
+
+		final List<String> tool_sections = Arrays.asList(
+			"DATASET_LABEL_PRINTING_PRESET", "LOT_LABEL_PRINTING_PRESET",
+			"GERMPLASM_LABEL_PRINTING_PRESET", "GERMPLASM_LIST_LABEL_PRINTING_PRESET",
+			"STUDY_ENTRIES_LABEL_PRINTING_PRESET");
+
+		this.jacksonMapper = new ObjectMapper();
+		this.jacksonMapper.disable(MapperFeature.DEFAULT_VIEW_INCLUSION);
+		JdbcConnection dbConn = null;
+		Statement selectStatement = null;
+		Statement updateStatement = null;
+
+		try {
+			dbConn = (JdbcConnection) database.getConnection();
+			dbConn.setAutoCommit(false);
+			selectStatement = dbConn.createStatement();
+			updateStatement = dbConn.createStatement();
+
+			ResultSet rs = selectStatement.executeQuery(
+				"SELECT * FROM program_preset where tool_section in (" +
+					tool_sections.stream().collect(Collectors.joining("','", "'", "'"))
+					+ ")");
+
+			while (rs.next()) {
+				final String program_preset_id = rs.getString("program_preset_id");
+				final String configuration = rs.getString("configuration");
+
+				try {
+					final LabelPrintingPresetDTO labelPrintingPresetDTO =
+						this.jacksonMapper.readValue(configuration, LabelPrintingPresetDTO.class);
+					final List<List<String>> selectedFields = new ArrayList<>();
+					labelPrintingPresetDTO.getSelectedFields().stream().forEach(list -> {
+						selectedFields.add(list.stream().map((fieldId) -> {
+							return this.concatenateFieldTypeName(fieldId);
+						}).collect(Collectors.toList()));
+					});
+
+					labelPrintingPresetDTO.setSelectedFields(selectedFields);
+					if (!labelPrintingPresetDTO.getBarcodeSetting().isAutomaticBarcode()) {
+						final List<String> barCodeFields = labelPrintingPresetDTO.getBarcodeSetting().getBarcodeFields().stream()
+							.map((fieldId) -> {
+								return this.concatenateFieldTypeName(fieldId);
+							}).collect(Collectors.toList());
+						labelPrintingPresetDTO.getBarcodeSetting().setBarcodeFields(barCodeFields);
+					}
+
+					final String UpdatedConfiguration =
+						this.jacksonMapper.writerWithView(PresetDTO.View.Configuration.class).writeValueAsString(labelPrintingPresetDTO);
+
+					final String updateSql = String.format(//
+						"update program_preset set configuration = '%s' where program_preset_id = %s", UpdatedConfiguration, program_preset_id);
```

Comment:
@cuenyad the SQL can be static to the class

Reply:
done!